tests/johnsnowlabs/test_jsl_api.py

```diff
@@ -0,0 +1,782 @@
+    # copy local dir to and back from HadoopFS and make sure the results match
+    from mlflow.spark import _HadoopFileSystem as FS
+    test_dir_0 = os.path.join(str(tmpdir), "expected")
+    test_file_0 = os.path.join(test_dir_0, "root", "file_0")
+    test_dir_1 = os.path.join(test_dir_0, "root", "subdir")
+    test_file_1 = os.path.join(test_dir_1, "file_1")
+    os.makedirs(os.path.dirname(test_file_0))
+    with open(test_file_0, "w") as f:
+        f.write("test0")
+    os.makedirs(os.path.dirname(test_file_1))
+    with open(test_file_1, "w") as f:
+        f.write("test1")
+    remote = "/tmp/mlflow/test0"
+    # File should not be copied in this case
+    assert os.path.abspath(test_dir_0) == FS.maybe_copy_from_local_file(test_dir_0, remote)
+    FS.copy_from_local_file(test_dir_0, remote, remove_src=False)
+    local = os.path.join(str(tmpdir), "actual")
+    FS.copy_to_local_file(remote, local, remove_src=True)
+    assert sorted(os.listdir(os.path.join(local, "root"))) == sorted(
+        ["subdir", "file_0", ".file_0.crc"]
+    )
+    assert sorted(os.listdir(os.path.join(local, "root", "subdir"))) == sorted(
+        ["file_1", ".file_1.crc"]
+    )
+    # compare the files
+    with open(os.path.join(test_dir_0, "root", "file_0")) as expected_f:
+        with open(os.path.join(local, "root", "file_0")) as actual_f:
+            assert expected_f.read() == actual_f.read()
+    with open(os.path.join(test_dir_0, "root", "subdir", "file_1")) as expected_f:
+        with open(os.path.join(local, "root", "subdir", "file_1")) as actual_f:
+            assert expected_f.read() == actual_f.read()
+
+    # make sure we cleanup
+    assert not os.path.exists(FS._remote_path(remote).toString())  # skip file: prefix
+    FS.copy_from_local_file(test_dir_0, remote, remove_src=False)
+    assert os.path.exists(FS._remote_path(remote).toString())  # skip file: prefix
+    FS.delete(remote)
+    assert not os.path.exists(FS._remote_path(remote).toString())  # skip file: prefix
+
+
+def test_model_export(jsl_model, model_path):
+    mlflow.johnsnowlabs.save_model(jsl_model, path=model_path)
+    # 1. score and compare reloaded sparkml model
+    reloaded_model = mlflow.johnsnowlabs.load_model(model_uri=model_path)
+    validate_model(jsl_model, reloaded_model)
+    # 2. score and compare reloaded pyfunc
+    validate_model(jsl_model, pyfunc.load_model(model_path))
+    # 3. score and compare reloaded pyfunc Spark udf
+    preds3 = score_model_as_udf(model_uri=model_path)
+    d1 = pd.DataFrame(json.loads(preds3[0])).drop(columns=['index']).reset_index().drop(columns=['index'])
+    d2 = jsl_model.predict('Hello world')
+    d1, d2 = fix_dataframe_with_respect_for_nlu_issues(d1, d2)
+    assert d1.equals(d2)
+    assert os.path.exists(MLFLOW_DFS_TMP.get())
+
+
+def test_model_export_qucik(jsl_model, model_path):
+    mlflow.johnsnowlabs.save_model(jsl_model, path='MY_MODEL3')
+    # 1. score and compare reloaded sparkml model
```

Comment:
Please finish this test fragment

Reply:
Thanks for the hint, this is irrelevant dropped it